clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy

```diff
@@ -114,28 +113,32 @@ class CatsSearchProvider implements SearchProvider, Runnable {
    */
   @Override
   void run() {
-    log.info("Refreshing Cached Identifiers (instances)")
-    def instanceIdentifiers = providers.findAll { provider ->
-      provider.supportsSearch('instances', Collections.emptyMap())
-    }.collect { provider ->
-      def cache = providerRegistry.getProviderCache(provider.getProviderName())
-      return cache.getIdentifiers("instances").findResults { key ->
-        def v = provider.parseKey(key)
-        if (v) {
-          v["_id"] = key
-        }
+    try {
+      log.info("Refreshing Cached Identifiers (instances)")
+      def instanceIdentifiers = providers.findAll { provider ->
+        provider.supportsSearch('instances', Collections.emptyMap())
+      }.collect { provider ->
+        def cache = providerRegistry.getProviderCache(provider.getProviderName())
+        return cache.getIdentifiers("instances").findResults { key ->
+          def v = provider.parseKey(key)
+          if (v) {
+            v["_id"] = key
+          }
 
-        return v?.collectEntries {
-          [it.key, it.value.toLowerCase()]
+          return v?.collectEntries {
+            [it.key, it.value.toLowerCase()]
+          }
         }
+      }.flatten()
+
+      if (instanceIdentifiers) {
+        cachedIdentifiersByType.set(["instances": instanceIdentifiers])
       }
-    }.flatten()
 
-    if (instanceIdentifiers) {
-      cachedIdentifiersByType.set(["instances": instanceIdentifiers])
+      log.info("Refreshed Cached Identifiers (found ${instanceIdentifiers.size()} instances)")
+    } catch (Exception e) {
+      log.error("Unable to refresh cached identifiers", e)
```

Comment:
You may want to include `(instances)` in the log message as you did at the beginning of the try/catch.